### PR TITLE
update localcluster deployment doc to use Docker desktop for local deployment Resolves issue #3819

### PR DESCRIPTION
**Pull request by ElayAharoni**
Resolves https://github.com/kubeflow/website/issues/3819

### CODE REVIEW

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 1]
```diff
@@ -311,10 +312,34 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
 For more information about K3ai, refer to the
 [official documentation](https://k3ai.github.io/docs/intro).
 
+## Docker-Desktop
+
+Docker Desktop is secure, out-of-the-box containerization software offering developers and teams a robust, hybrid toolkit to build, share, and run applications anywhere.
+
+### 1. installing Docker-Desktop
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:319]:
```suggestion
### 1. Installing Docker-Desktop
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 2]
```diff
@@ -311,10 +312,34 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+
+You can install and configure Docker-Desktop by following the
+[official quick start](https://www.docker.com/products/docker-desktop/).
+
+**on Windows** - download and run the Docker Desktop Installer.exe file, and follow the instructions inside the installer.
+
+### 2. Creating a cluster on Docker-Desktop
+
+Having installed Docker-Desktop, you can create a Kubernetes cluster on Docker-Desktop by following those steps:
+1. open the Docker-Desktop dashboard.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:329]:
```suggestion
1. Open the Docker-Desktop dashboard.
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 3]
```diff
@@ -311,10 +312,34 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+2. open the settings by pressing the settings button.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:330]:
```suggestion
2. Open the settings by pressing the settings button.
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 4]
```diff
@@ -311,10 +312,34 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+3. navigating to the Kubernetes tab on the left side.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:331]:
```suggestion
3. Navigate to the Kubernetes tab on the left side.
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 5]
```diff
@@ -311,10 +312,34 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+4. checking the 'Enable Kubernetes' checkbox, and pressing on 'Apply and restart'. and wait for the app to restart.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:332]:
```suggestion
4. Check the 'Enable Kubernetes' checkbox, click 'Apply and restart' and wait for the app to restart.
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 6]
```diff
@@ -311,10 +312,34 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+5. in order to check if the cluster is up and running, open the the command prompt and run the following command: "kubectl cluster-info" and you should see the following output:
+   "Kubernetes control plane is running at https://kubernetes.docker.internal:<some-port>
+   CoreDNS is running at https://kubernetes.docker.internal:<some-port>/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:335]:
```suggestion
5. In order to check if the cluster is up and running, open the command prompt and run the following command:
    ```SHELL
    kubectl cluster-info   
    ```
    You should see the following output:
    ```SHELL
    Kubernetes control plane is running at https://kubernetes.docker.internal:<some-port>
    CoreDNS is running at https://kubernetes.docker.internal:<some-port>/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
    ```
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 7]
```diff
@@ -311,10 +312,34 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+
+To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:337]:
```suggestion
To further debug and diagnose cluster problems, use `kubectl cluster-info dump`.
```

**Review by hbelmiro — COMMENTED** [review]
Thank you @ElayAharoni.
I left a few suggestions.

/ok-to-test

**Comment by hbelmiro** [comment]
> solved this issue:
> add local deployment documentation for pipelines with Docker-Desktop #3819

Also, can you please change the PR description to:

```
Resolves https://github.com/kubeflow/website/issues/3819.
```

That way GitHub will close the issue automatically when this PR gets merged.

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md (hunk 7 above)
Comment (ElayAharoni) [reply content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md]: thanks for your comments, i took your advices and made the changes

**Comment by hbelmiro** [comment]
@ElayAharoni you misunderstood me. You've changed the title. I meant the description.
👇🏽 
Screenshot from 2024-07-22 14-37-28

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 8]
```diff
@@ -311,10 +312,40 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+You can install and configure Docker-Desktop by following the
+[official quick start](https://www.docker.com/products/docker-desktop/).
+
+**on Windows** - download and run the Docker Desktop Installer.exe file, and follow the instructions inside the installer.
+
+### 2. Creating a cluster on Docker-Desktop
+
+Having installed Docker-Desktop, you can create a Kubernetes cluster on Docker-Desktop by following those steps:
+1. Open the Docker-Desktop dashboard.
+2. Open the settings by pressing the settings button.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:330]:
```suggestion
2. Open the settings by clicking the settings button.
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 9]
```diff
@@ -311,10 +312,40 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+3. Navigating to the Kubernetes tab on the left side.
+4. Checking the 'Enable Kubernetes' checkbox, and pressing on 'Apply and restart'. and wait for the app to restart.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:332]:
```suggestion
4. Check the 'Enable Kubernetes' checkbox, click 'Apply and restart' and wait for the app to restart.
```

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 10]
```diff
@@ -311,10 +312,40 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+[official quick start](https://www.docker.com/products/docker-desktop/).
+
+**on Windows** - download and run the Docker Desktop Installer.exe file, and follow the instructions inside the installer.
+
+### 2. Creating a cluster on Docker-Desktop
+
+Having installed Docker-Desktop, you can create a Kubernetes cluster on Docker-Desktop by following those steps:
+1. Open the Docker-Desktop dashboard.
+2. Open the settings by pressing the settings button.
+3. Navigating to the Kubernetes tab on the left side.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:331]:
```suggestion
3. Navigate to the Kubernetes tab on the left side.
```

**Review by hbelmiro — COMMENTED** [review]
@ElayAharoni There are still some suggestions that were not addressed.

**Comment by ElayAharoni** [comment]
thanks, i have made the suggested changes

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 11]
```diff
@@ -311,10 +312,41 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+**on Windows** - 
+download and run the Docker Desktop Installer.exe file, and follow the instructions inside the installer.
+
+### 2. Creating a cluster on Docker-Desktop
+
+Having installed Docker-Desktop, you can create a Kubernetes cluster on Docker-Desktop by following those steps:
+1. Open the Docker-Desktop dashboard.
+2. Open the settings by clicking the settings button.
+3. Navigate to the Kubernetes tab on the left side.
+4. check the 'Enable Kubernetes' checkbox, click on 'Apply and restart'. and wait for the app to restart.
```
Comment (hbelmiro) [inline content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md:333]:
```suggestion
4. Check the 'Enable Kubernetes' checkbox, click 'Apply and restart' and wait for the app to restart.
```

**Review by hbelmiro — COMMENTED** [review]
@ElayAharoni one last suggestion that you've missed...

File: content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md [hunk 12]
```diff
@@ -311,10 +312,41 @@ curl -sfL https://get.k3ai.in | bash -s -- --gpu --plugin_kfpipelines
...
+**on Windows** - 
+download and run the Docker Desktop Installer.exe file, and follow the instructions inside the installer.
+
+### 2. Creating a cluster on Docker-Desktop
+
+Having installed Docker-Desktop, you can create a Kubernetes cluster on Docker-Desktop by following those steps:
+1. Open the Docker-Desktop dashboard.
+2. Open the settings by clicking the settings button.
+3. Navigate to the Kubernetes tab on the left side.
+4. Check the 'Enable Kubernetes' checkbox, click on 'Apply and restart'. and wait for the app to restart.
```
[the remark this answers is not in the document]
Comment (ElayAharoni) [reply content/en/docs/components/pipelines/legacy-v1/installation/localcluster-deployment.md]: i am sorry didn't noticed those.. fixed them now

**Review by hbelmiro — APPROVED** [review]
/lgtm

**Pull request merged**